buchschloss/gui2/main.py

```diff
@@ -215,55 +168,87 @@ def filler(event=None):
     isbn_field.bind('<FocusOut>', filler)
 
 
+def get_actions(spec):
+    """return a dict of actions suitable for ActionTree.from_map"""
+    wrapped_action_ns = {
+        k: common.NSWithLogin(getattr(core, k))
+        for k in ('Book', 'Person', 'Group', 'Library', 'Borrow', 'Member', 'Script')
+    }
+    default_action_adapters = {
+        'new': lambda name, ns: generic_formbased_action('new', get_form(name), ns.new),
+        'edit': lambda name, ns: generic_formbased_action(
+            'edit', get_form(name), ns.edit, fill_data=ns.view_ns),
+        'view': lambda name, __: ShowInfo.instances.get(name),
+        'search': lambda name, ns: actions.search(
+            get_form(name + '_search', get_form(name)), ns.search, ShowInfo.instances[name]),
+    }
+    special_action_funcs = {
+        ('Group', 'edit'): generic_formbased_action(
+            'edit', forms.GroupForm, wrapped_action_ns['Group'].edit),
+        ('Library', 'edit'): generic_formbased_action(
+            'edit', forms.LibraryForm, wrapped_action_ns['Library'].edit),
+        ('Group', 'activate'): generic_formbased_action(
+            None, forms.GroupActivationForm, wrapped_action_ns['Group'].activate),
+        ('Member', 'change_password'): generic_formbased_action(
+            None, forms.ChangePasswordForm, wrapped_action_ns['Member'].change_password),
+        ('Borrow', 'restitute'): generic_formbased_action(
+            None, forms.RestituteForm, wrapped_action_ns['Borrow'].restitute),
+        ('Book', 'new'): generic_formbased_action(
+            'new', forms.BookForm, actions.new_book, post_init=new_book_autofill),
+    }
+
+    def get_form(name, *default):
+        return getattr(forms, name.title().replace('_', '') + 'Form', *default)
```

Comment:
is the snake to camel actually use anywhere except from search default action, where it can be instantly replaced?